modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkUtils.java

```diff
@@ -0,0 +1,551 @@
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return noOfLinesExecuted;
+    }
+
+    public static LocalTime getCurrentTimeStamp() throws InterruptedException {
+
+        Thread.sleep(15000);
+        LocalDateTime ldt = LocalDateTime.now();
+        LocalTime currentTime = LocalTime.parse(DateTimeFormatter.ofPattern("HH:mm:ss", Locale.ENGLISH).format(ldt));
+        Thread.sleep(1000);
+        return currentTime;
+    }
+
+    public static void validateBenchmark(int benchmark, int actualCount) {
+
+        boolean exceedsLimit = false;
+
+        benchmark = (int) (benchmark + (benchmark * 0.01));
+//        Validate if No. of statements executed exceeds the benchmark
+        if (actualCount > benchmark) {
+            exceedsLimit = true;
+        }
+        assertFalse(
+                "Exceeds limit! " + actualCount + " were Executed, but the Benchmark value is " + benchmark,
+                exceedsLimit);
+    }
+
+    public static String getSystemResourceLocation() {
+
+        String resourceLocation;
+        if (System.getProperty("os.name").toLowerCase().contains(OperatingSystems.WINDOWS.name().toLowerCase())) {
+            resourceLocation = System.getProperty("framework.resource.location").replace("/", "\\");
+        } else {
+            resourceLocation = System.getProperty("framework.resource.location").replace("/", "/");
+        }
+
+        return resourceLocation;
+    }
+
+    public static int getBenchmark(String testType, String scenario) throws IOException, ParseException {
+
+        String benchmarkValue = null;
+        String resourceLocation = getSystemResourceLocation() + "benchmark-values" + File.separator + "benchmark-values-" + testType + ".json";
+
+        JSONParser parser = new JSONParser();
+        JSONArray a = (JSONArray) parser.parse(new FileReader(resourceLocation));
+
+        for (Object o : a) {
+            JSONObject values = (JSONObject) o;
+            benchmarkValue = (String) values.get(scenario);
+        }
+        return Integer.parseInt(benchmarkValue);
+    }
+
+    public static void writeResultsToFile(String fileName, String testName, int actual, int benchmark) throws IOException {
+
+        String outputFile = OUT_FILE_PATH + "Results_" + fileName + ".log";
+        File f = new File(outputFile);
+        if (!f.exists()) {
+            f.createNewFile();
+        }
+        BufferedWriter bw = new BufferedWriter(new FileWriter(f, true));
+        bw.append(testName + " ===== Actual count is " + actual + "   Benchmark is  " + benchmark);
+        bw.newLine();
+        bw.close();
+    }
+
+    public void generateConsumerCredentialsAndAccessToken() throws IOException {
```

Comment:
why do we need non-static methods in a util class?

Reply:
they should be static. made the changes.